### PR TITLE
splunkattackanalyzer: Feature - Expose user agent, archive password and exit region submission parameters

### PR DESCRIPTION
**Pull request by dfederschmidt**
This PR adds optional parameters to file and url submission to specify user agent, exit region for Web Analyzer runs and archive_password for extracting password-protected archives

Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [x] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [x] If this is a new connector or you are adding new actions
    - [x] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [x] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [x] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [x] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [x] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-t
hrough-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
- Added optional parameters to file and url submission to specify user agent, exit region for Web Analyzer runs and archive_password for extracting password-protected archives

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by ishans-crest** [comment]
Hi, @dfederschmidt 
We have observed that the new mandatory parameter user_agent will cause a backward compatibility issue. This is because - the existing playbooks using submit_url or submit_file actions will fail as they will not provide the user-agent value which now is mandatory. 

Now, because this parameter is not mandatory from the API side, we can also have this parameter as optional in the soar connector. 
Should we make it an Optional parameter? 
is there a reason to keep it mandatory?

**Comment by dfederschmidt** [comment]
@ishans-crest 

Thanks for your thoughts. You're right. 

There is no pressing need to make this a mandatory parameter. If not provided, we can do a fallback to the default user agent value. Would you want me to make this change?

> Hi, @dfederschmidt We have observed that the new mandatory parameter user_agent will cause a backward compatibility issue. This is because - the existing playbooks using submit_url or submit_file actions will fail as they will not provide the user-agent value which now is mandatory.
> 
> Now, because this parameter is not mandatory from the API side, we can also have this parameter as optional in the soar connector. Should we make it an Optional parameter? is there a reason to keep it mandatory?

**Comment by ishans-crest** [comment]
Hi @dfederschmidt. Thanks for the reply. Regarding the changes, - we will make the necessary changes including making the user-agent an optional parameter.

File: .gitignore
Comment (tonyc-phantom) [inline .gitignore:1]: Are you supposed to commit this file?

**Review by tonyc-phantom — DISMISSED** [review]

**Review by splunk-jessica — APPROVED** [review]

File: .gitignore
Comment (ishans-crest) [reply .gitignore:1]: Removed .(dot) files.

**Review by ishans-crest — APPROVED** [review]

**Pull request merged**